segmentation/include/pcl/segmentation/impl/region_growing.hpp

```diff
@@ -610,17 +609,17 @@ pcl::RegionGrowing<PointT, NormalT>::getSegmentFromPoint (int index, pcl::PointI
     }
     // if we have already made the segmentation, then find the segment
     // to which this point belongs
-    for (auto i_segment = clusters_.cbegin (); i_segment != clusters_.cend (); i_segment++)
+    for (const auto& i_segment : clusters_)
     {
       bool segment_was_found = false;
-      for (std::size_t i_point = 0; i_point < i_segment->indices.size (); i_point++)
+      for (const auto& i_point : (i_segment.indices))
       {
-        if (i_segment->indices[i_point] == index)
+        if (i_point == index)
         {
           segment_was_found = true;
           cluster.indices.clear ();
-          cluster.indices.reserve (i_segment->indices.size ());
-          std::copy (i_segment->indices.begin (), i_segment->indices.end (), std::back_inserter (cluster.indices));
+          cluster.indices.reserve (i_segment.indices.size ());
+          std::copy (i_segment.indices.begin (), i_segment.indices.end (), std::back_inserter (cluster.indices));
```

Comment:
Should be rewritten using `find`, but not in scope of this PR

Reply:
You mean like this? `if (std::find (i_segment.indices.cbegin (), i_segment_indices.cend (), index) != i_segment.indices.cend ())`

Reply:
(In 2 lines, since we need to use the found iterator, but essentially) Yes. Looking at find and then an operation is a tad more understandable than seeing a loop, a comparison, a break and a lack of other operations.

Compare the current version with:
```cpp
const auto it =std::find (i_segment.indices.cbegin (), i_segment_indices.cend (), index);
segment_was_found = (it != i_segment_indices.cend());
if (segment_was_found) { // do the operation here
}
```